### ISSUE TITLE
ktlint inside detekt does not report issues that ktlint by itself does

### ISSUE THREAD

**Issue by goncalossilva** [detekt/detekt#4167]
This is similar to #2161, but it's been over 2 years since that one and it's supposed to be fixed.

## Expected Behavior
Running ktlint from inside detekt reports the same errors as running ktlint in isolation.

## Observed Behavior
ktlint errors are absent. I've been struggling with a Gradle config thinking the problem was there, but heading to CLI shows the same exact behavior:

```
$ detekt -p detekt-formatting-1.18.1.jar -i .
formatting - 20min debt
	StringTemplate - [] at /home/g/Code/D/ffs/build.gradle.kts:30:30
	NoTrailingSpaces - [] at /home/g/Code/D/ffs/library-backend/build.gradle.kts:37:1
	NoTrailingSpaces - [] at /home/g/Code/D/ffs/core/build.gradle.kts:38:1
	NoTrailingSpaces - [] at /home/g/Code/D/ffs/library-frontend/build.gradle.kts:37:1
style - 45min debt
	UnusedPrivateMember - [commonTest] at /home/g/Code/D/ffs/library-backend/build.gradle.kts:39:9
	UnusedPrivateMember - [jvmTest] at /home/g/Code/D/ffs/library-backend/build.gradle.kts:45:9
	UnusedPrivateMember - [jsTest] at /home/g/Code/D/ffs/library-backend/build.gradle.kts:51:9
	UnusedPrivateMember - [commonTest] at /home/g/D/Doist/ffs/core/build.gradle.kts:40:9
	UnusedPrivateMember - [jvmTest] at /home/g/Code/D/ffs/core/build.gradle.kts:46:9
	UnusedPrivateMember - [jsTest] at /home/g/Code/D/ffs/core/build.gradle.kts:52:9
	UnusedPrivateMember - [commonTest] at /home/g/Code/D/ffs/library-frontend/build.gradle.kts:39:9
	UnusedPrivateMember - [jvmTest] at /home/g/Code/D/ffs/library-frontend/build.gradle.kts:45:9
	UnusedPrivateMember - [jsTest] at /home/g/Code/D/ffs/library-frontend/build.gradle.kts:51:9

Overall debt: 1h 5min

Build failed with 13 weighted issues.
$
$ ktlint **/*.kt
/home/g/Code/D/ffs/dashboard/src/main/kotlin/Client.kt:8:1: Unexpected indentation (13) (should be 12)
/home/g/Code/D/ffs/server/src/main/kotlin/Router.kt:1:1: Imports must be ordered in lexicographic order without any empty lines in-between with "java", "javax", "kotlin" and aliases in the end
```

## Steps to Reproduce

The output above is from [this repo](https://github.com/goncalossilva/ffs) (minus the indentation error which is artificially introduced, but the import one is there). The Gradle setup [is done here](https://github.com/goncalossilva/ffs/blob/main/build.gradle.kts#L27-L47), and it might be incorrect, but I'd advise testing via CLI to keep things simple.

1. Run `ktlint **/*.kt`
2. Run `detekt --build-upon-default-config --plugins detekt-formatting-1.18.1.jar -i .`
3. Compare.

## Your Environment
* Version of detekt used: 1.18.1
* Version of ktlint used: 0.42.1
* Version of Gradle used (if applicable): 7.2
* Operating System and version: Fedora 34
* Link to your project (if it's a public repository): https://github.com/goncalossilva/ffs

**Comment by goncalossilva** [comment]
You're right! I assumed that by enabling ktlint, I'd get all of ktlint's default rules (its slogan is “no configuration” after all :sweat_smile:) but I can see that is not the case. Setting them in detekt.yml fixes this. I wonder if a note on the documentation about this is warranted? Thank you. :)

**Comment by 3flex** [comment]
I'll reopen - the default enabled formatting rules in detekt should match the ktlint defaults, so we should review and update where there's a discrepancy.
